lib/licensed/sources/swift.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+require "json"
+require "pathname"
+require "licensed/sources/helpers/content_versioning"
+
+module Licensed
+  module Sources
+    class Swift < Source
+      include Licensed::Sources::ContentVersioning
+
+      class Dependency < Licensed::Dependency
+        attr_reader :url
+
+        def initialize(name:, url:, version:, path:, search_root: nil, errors: [], metadata: {})
+          @url = url
+          super name: name, version: version, path: path, errors: errors, metadata: metadata, search_root: search_root
+        end
+      end
+
+      def enabled?
+        Licensed::Shell.tool_available?("swift") && swift_package?
+      end
+
+      def enumerate_dependencies
+        Hash[declarations.map { |d| [d.url, d] }]
+          .merge(Hash[pins.map { |d| [d.url, d] }])
+          .values
+      end
+
+      private
+
+      def declarations
+        return @declarations if defined?(@declarations)
+
+        @declarations = begin
+          json = JSON.parse(dump_package_command)
+          json["dependencies"].map do |dependency|
+            Dependency.new(
+              name: dependency["name"],
+              url: dependency["url"],
+              path: "/Package.swift",
```

Comment:
do swift packages not have unique paths on disk? or maybe the better question is, do swift packages, when brought into the local environment, contain license files or other legal contents as part of the package?

this tool generally expects license contents to be available on the local system, at the location given by this path argument

Reply:
> this tool generally expects license contents to be available on the local system, at the location given by this path argument

Ah, that's my mistake, then. I didn't understand the significance of the `path` parameter, and simply hardcoded the value.

When Swift Package Manager builds a package, it downloads each node in the resolved dependency graph into a subdirectory of `.build`. Each package is downloaded by performing a `git clone` on the dependency's URL. So we could determine the license associated with each dependency by looking in the root directory of each cloned repository.

Reply:
awesome, that makes sense.  yup just point path here at the packages git root 👍